### PR TITLE
Logs node e2e perf data to standalone json files

### PR DESCRIPTION
**Pull request by yguo0905**
Fixes the node-dash-perf issue in https://github.com/kubernetes/kubernetes/issues/44003.

- Move perf data types to `test/e2e/perftype/perftype.go` so that the node-perf-dash can depend on.
- Logs the perf data to standalone json files so that node-perf-dash can consume it easily. A sample run of `ci-kubernetes-node-kubelet-benchmark` is at https://console.cloud.google.com/storage/browser/ygg-gke-dev-bucket/e2e-node-test/ci-kubernetes-node-kubelet-benchmark/1.

The corresponding changes in node-perf-dash is at https://github.com/kubernetes/contrib/pull/2628.

**Release note**:
`None`

/sig node
/area node-e2e
/assign @Random-Liu

### CODE REVIEW

File: test/e2e/perftype/perftype.go [hunk 1]
```diff
@@ -18,6 +18,11 @@ package perftype
 
 // TODO(random-liu): Replace this with prometheus' data model.
 
+// CurrentKubeletPerfMetricsVersion is the current kubelet performance metrics
+// version. This is used by mutiple perf related data structures. We should
+// bump up the version each time we make an incompatible change to the metrics.
+const CurrentKubeletPerfMetricsVersion = "v2"
```
Comment (Random-Liu) [inline test/e2e/perftype/perftype.go]: Don't understand why we need to expose the version.

Comment (yguo0905) [inline test/e2e/perftype/perftype.go]: This is also used in node e2e test. See line 34 at `test/e2e_node/benchmark_util.go`.

Comment (Random-Liu) [inline test/e2e/perftype/perftype.go]: For internal use, I think it's fine to put it in perf_util.
`e2e_node` uses the same test framework with e2e.

**Comment by Random-Liu** [comment]
LGTM except the comment.

File: test/e2e/perftype/perftype.go (hunk 1 above)
Comment (yguo0905) [inline test/e2e/perftype/perftype.go]: OK, done.

**Comment by yguo0905** [comment]
@k8s-bot pull-kubernetes-e2e-gce-etcd3 test this

**Comment by dchen1107** [comment]
/lgtm

But the pr doesn't need to be merged into 1.7 branch anyway, right?

**Comment by yguo0905** [comment]
@dchen1107, right, it doesn't need to be merged into 1.7 branch since the tests are running against master.

**Comment by dchen1107** [comment]
/lgtm

I read the pr more. This fixes node perf dashboard, which helps we understand the new release's kubelet & docker resource usages, and no risk for release itself.

**Comment by yguo0905** [comment]
@k8s-bot pull-kubernetes-kubemark-e2e-gce test this

**Comment by yguo0905** [comment]
@k8s-bot pull-kubernetes-e2e-gce-etcd3 test this

**Comment by yguo0905** [comment]
Not sure why this PR was merged automatically before all tests passed.

@k8s-bot pull-kubernetes-e2e-gce-etcd3 test this

**Pull request merged**